### PR TITLE
Fix various Clippy lints

### PR DESCRIPTION
**Pull request by SpecificProtagonist**
Fixed various warnings shown by Clippy (a linter for Rust).
- If you want to integrate Clippy in VS Codium, use `"rust-analyzer.check.command": "clippy"`;
  in Helix, use `checkOnSave = {command = "clippy"}` (in languages.toml → rust)
- When all warnings are fixed, it is probably a good idea to [run it in CI & fail on warnings](https://doc.rust-lang.org/clippy/continuous_integration/github_actions.html)
- Haven't touched unused symbols as I don't know which ones are superfluous (delete) and which ones are still todo (prefix with _)
- I've allowed [`clippy::module_inception`](https://rust-lang.github.io/rust-clippy/master/index.html#/module_inception), but some renaming (such as `guilds.rs` → `guild.rs`) might make things a little more idiomatic. Mostly bikeshedding though.
- Is there a reason `MessageSendSchema`'s fields aren't all pub? If so, I'd turn it into a builder like with `RegisterSchema`, otherwise I'd just make them pub.
- There are a lot of [`clippy::await_holding_refcell_ref`](https://rust-lang.github.io/rust-clippy/master/index.html#/await_holding_refcell_ref) warnings. Most of those are false positives. I didn't want to disable this lint though because they are easily fixed:
  ```diff
  - let belongs_to = user.belongs_to.borrow();
  - let url_api = belongs_to.urls.get_api().to_string();
  - drop(belongs_to);
  +let url_api = {
  +    let belongs_to = user.belongs_to.borrow();
  +    belongs_to.urls.get_api().to_string()
  +};
  ```
  and because there are true positives, such as in `src/api/common.rs:15`. I'm not very familiar with Chorus nor with async Rust in general, so I don't know what's best in these cases, but it might be worth removing interior mutability from `UserMeta::belongs_to` and adding it to `Instance::limits`.

### CODE REVIEW

**Comment by bitfl0wer** [comment]
A few comments on this:

>     * When all warnings are fixed, it is probably a good idea to [run it in CI & fail on warnings](https://doc.rust-lang.org/clippy/continuous_integration/github_actions.html)

I am not sure i'd like to integrate it into the main Build & Test run, but I am probably going to make a seperate CI action to run clippy in CI.

>     * I've allowed [`clippy::module_inception`](https://rust-lang.github.io/rust-clippy/master/index.html#/module_inception), but some renaming (such as `guilds.rs` → `guild.rs`) might make things a little more idiomatic. Mostly bikeshedding though.

That might be a good idea. I don't know if I'd like to tackle this task myself right now though, but PRs are always welcome :)

>     * Is there a reason `MessageSendSchema`'s fields aren't all pub? If so, I'd turn it into a builder like with `RegisterSchema`, otherwise I'd just make them pub.

No. There isn't a reason xD. A thing you have to understand is, that Chorus is my first Rust project, and while I am generally happy with its' architecture, the oldest code in the project definitely lacks some idiomaticity. :) 

>     * There are a lot of [`clippy::await_holding_refcell_ref`](https://rust-lang.github.io/rust-clippy/master/index.html#/await_holding_refcell_ref) warnings. Most of those are false positives. I didn't want to disable this lint though because they are easily fixed:
>       ```diff
>       - let belongs_to = user.belongs_to.borrow();
>       - let url_api = belongs_to.urls.get_api().to_string();
>       - drop(belongs_to);
>       +let url_api = {
>       +    let belongs_to = user.belongs_to.borrow();
>       +    belongs_to.urls.get_api().to_string()
>       +};
>       ```     
>       and because there are true positives, such as in `src/api/common.rs:15`. 

The suggested refactor looks good :) Didn't know that it could be done this way, because, as I said, it's my first two months with the language. Will look at this though! Thanks for bringing it up.

> I'm not very familiar with Chorus nor with async Rust in general, so I don't know what's best in these cases, but it might be worth removing interior mutability from `UserMeta::belongs_to` and adding it to `Instance::limits`.

What would be the possible benefits of doing it this way? Interior Mutability has to be used either way afaik, and currently, the relationship between a User and the Instance they belong to works fine and is, hopefully, pretty clear to anyone who has to work with this code. I would be happy to hear your thoughts on this.

File: src/gateway.rs [hunk 1]
```diff
@@ -388,14 +365,10 @@ impl Gateway {
             let msg = self.websocket_receive.next().await;
 
             // This if chain can be much better but if let is unstable on stable rust
```
Comment (kozabrada123) [inline src/gateway.rs:367]: This comment is no longer relevant

**Comment by kozabrada123** [comment]
Looks good :D 
[You need to derive default for MessageSendSchema though to fix the failing build though](https://github.com/polyphony-chat/chorus/actions/runs/5311801270/jobs/9615603714?pr=109#step:6:233)

**Review by bitfl0wer — APPROVED** [review]

**Comment by bitfl0wer** [comment]
Added a clippy workflow and fixed the failing build, but LGTM other than that! Thank you for your contribution. :)

**Pull request merged**